tensorflow/core/kernels/fixed_length_record_reader_op.cc

```diff
@@ -62,14 +64,20 @@ class FixedLengthRecordReader : public ReaderBase {
 
   Status ReadLocked(string* key, string* value, bool* produced,
                     bool* at_end) override {
-    if (input_buffer_->Tell() >= file_pos_limit_) {
+    if (input_buffer_->Tell() >= file_pos_limit_ ||
+        input_buffer_->Tell() + record_bytes_ > file_pos_limit_) {
```

Comment:
For the original `FixedLengthRecordReader`, I agree with you that the caller is possible and is responsible for setting up everything for the reader. 
But I don't think it's true for the condition `hop_bytes>0`. Under this situation, it maybe not possible for caller to setting up everything. Consider following condition.

File:
`0123456`

Reader setting:
`record_bytes=3`, `hop_bytes=2`, `footer_bytes=0`, `header_bytes=0`

What we expected is:
`012`, `234`, `456`

If I get it right, with the original checking condition(`input_buffer_->Tell() >= file_pos_limit_`), what we actually get will be:
`012`, `234`, `456`, `Error`.
There seems not possible to avoid the last error, because for the last record, the remaining bytes will always be less than `record_bytes`: `01234|56`("|" for the start position of the last record)

Or should errors like this be catched by the caller?